src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -433,6 +433,56 @@ const char *smbios_mainboard_sku(void)
 	return sku;
 }
 
+int fill_mainboard_smbios_type16(unsigned long *current, int *handle)
+{
+	u8 spd_index = get_spd_offset();
+	u8 spd_buffer[CONFIG_DIMM_SPD_SIZE];
+	if (read_ddr3_spd_from_cbfs(spd_buffer, spd_index) < 0) {
+		return 0;
+	}
+
+	struct smbios_type16 *t = (struct smbios_type16 *)*current;
+	int len = sizeof(struct smbios_type16);
+	memset(t, 0, sizeof(struct smbios_type16));
+
+	t->handle = *handle;
+	t->length = len = sizeof(*t) - 2;
+	t->type = SMBIOS_PHYS_MEMORY_ARRAY;
+	
+	switch (spds[0][4])
+	{
+		case 0x03:
+			t->maximum_capacity = 2 * 1024 * 1024; // 2GB (in kB)
```

Comment:
Note that the type16 describes the memory controller capabilities rather than populated memory configuration. The maximum capacity should be the total number of GBs that are supported by the processor. In this case, I think we can cut the maximum capacity to 4GBs, since the motherboard cannot support more due to its design.